src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -579,49 +532,92 @@ impl A320Hydraulic {
             overhead_panel,
             &self.forward_cargo_door,
             &self.aft_cargo_door,
-            self.yellow_loop.is_pressurised(),
+            self.yellow_loop.system_section_switch_pressurised(),
         );
         self.yellow_electric_pump.update(
             context,
-            &self.yellow_loop,
+            self.yellow_loop.system_pressure(),
+            self.yellow_loop.reservoir(),
             &self.yellow_electric_pump_controller,
         );
 
-        self.ram_air_turbine
-            .update(context, &self.blue_loop, &self.ram_air_turbine_controller);
+        self.ram_air_turbine.update(
+            context,
+            self.blue_loop.system_pressure(),
+            self.blue_loop.reservoir(),
+            &self.ram_air_turbine_controller,
+        );
 
         self.green_loop_controller.update(engine_fire_push_buttons);
         self.green_loop.update(
-            context,
-            Vec::new(),
-            vec![&self.engine_driven_pump_1],
-            Vec::new(),
-            vec![&self.power_transfer_unit],
+            &mut vec![&mut self.engine_driven_pump_1],
+            &mut None,
+            &Some(&self.power_transfer_unit),
+            &context,
             &self.green_loop_controller,
         );
 
         self.yellow_loop_controller.update(engine_fire_push_buttons);
         self.yellow_loop.update(
+            &mut vec![&mut self.engine_driven_pump_2],
+            &mut Some(&mut self.yellow_electric_pump),
+            &Some(&self.power_transfer_unit),
             context,
-            vec![&self.yellow_electric_pump],
-            vec![&self.engine_driven_pump_2],
-            Vec::new(),
-            vec![&self.power_transfer_unit],
             &self.yellow_loop_controller,
         );
+        // println!(
+        //     "Yloop P={:.0} Gloop={:.0} Yedp={:.0} Gedp={:.0}",
+        //     self.yellow_loop.system_pressure().get::<psi>(),
+        //     self.green_loop.system_pressure().get::<psi>(),
+        //     self.yellow_loop.pump_pressure(0).get::<psi>(),
+        //     self.green_loop.pump_pressure(0).get::<psi>(),
+        // );
+        // use systems::hydraulic::PressureSource;
+        // println!(
+        //     "Bloop P={:.0} Bepump={:.0} bdisp={:.2}",
+        //     self.blue_loop.system_pressure().get::<psi>(),
+        //     self.blue_loop.pump_pressure(0).get::<psi>(),
+        //     self.blue_electric_pump.displacement().get::<cubic_inch>(),
+        // );
```

Comment:
Are these still needed?

Reply:
I might use it again before going review state, but I can totally forget to remove this in the end indeed :D